rdr_service/resource/generators/participant.py

```diff
@@ -770,18 +775,21 @@ def _calculate_enrollment_status(self, summary):
                         else:
                             gror_date_range.add_stop(response_date)
 
-                date_overlap = study_consent_date_range\
-                    .get_intersection(pm_date_range)\
-                    .get_intersection(baseline_modules_date_range)\
-                    .get_intersection(dna_date_range)\
-                    .get_intersection(ehr_date_range)
-
-                if summary['consent_cohort'] == BQConsentCohort.COHORT_3.name:
-                    date_overlap = date_overlap.get_intersection(gror_date_range)
-
-                # If there's any time that they had everything at once, then they should be a Core participant
-                if date_overlap.any():
-                    status = EnrollmentStatusV2.CORE_PARTICIPANT
+                try:
+                    date_overlap = study_consent_date_range\
+                        .get_intersection(pm_date_range)\
+                        .get_intersection(baseline_modules_date_range)\
+                        .get_intersection(dna_date_range)\
+                        .get_intersection(ehr_date_range)
+
+                    if summary['consent_cohort'] == BQConsentCohort.COHORT_3.name:
+                        date_overlap = date_overlap.get_intersection(gror_date_range)
+
+                    # If there's any time that they had everything at once, then they should be a Core participant
+                    if date_overlap.any():
+                        status = EnrollmentStatusV2.CORE_PARTICIPANT
+                except TypeError:
+                    pass
```

Comment:
What error happened that this solves? There's an 'if' check at the start of working with the date ranges that I thought implies that anything here would be available.

Reply:
Apparently None values can creep into the data sent to the 'get_intersection()' function which causes a TypeError exception.  My thinking with the exception handling here is; that if the participant doesn't have enough data and an exception is thrown, then they probably won't be made a Core Participant and its Ok to just move on.

Reply:
Keep in mind, these errors are from the Stable environment which has some pretty terrible data there. It may be that in Prod this exception will never be thrown.

Reply:
I have wondered about these get_intersection() use cases.   I'm new to this but is it a problem that we set add_start() but not add_stop()  to the ranges sometimes (often)?

Reply:
No, I built them out to be able to handle un-ending timeframes. Using consent for study as an example: to add a start but no end implies that consent was given and never revoked. I'm not sure, but I'm guessing these errors came up when None was added for start dates.

Robert and I talked in the slack group and he's adding a log message in case we see this in prod.